department-of-veterans-affairs/caseflow
# Resend initial letter from send final task

Resolves APPEALS-13373

Description
Create Resend initial notification letter from Send final notification letter task. In the Clerk of the board view the Send Final Notification Letter task cases are under Unassigned tab. If you click on the Case details, in the Currently active task you can see the task as Send Final Notification Letter Task. In the Currently active task, from Actions dropdown selection you need to select Resend initial notification letter. When you click on the case and go to Case details you can see the task as Send Final Notification Letter Task. In the Case timeline you can see the Send Final Notification Letter Completed.

Acceptance Criteria
- [x] The following occurs when the “Resend initial notification letter” task action is 
       selected for the Send Final Notification Letter task (used when the initial notification 
       letter needs to be resent as the prior contained an error) 
      Modal appears asking user to confirm that they want to resend initial notification 
      letter
      User is required to provide a reason why the letter needs to be resent
      Field type: free text with 350-character limit
- [x] Once confirmed:
       Send Final Notification Letter task is cancelled
       Task moves from Currently Active Tasks section to Case Timeline and contains the 
       following information
       Date of occurrence
       Assigned to: user or organization
       Assigned by: user who completed action to assign task to COB
       Cancelled by: user who completed Send Final Notification Letter task
       Task Instructions: user input in the Resent initial notification letter modal
 A new Send Initial Notification Letter task is created and active
       Task appears within the Currently Active Tasks section
       Task is assigned to the Clerk of the Board organization
       Case appears in the Unassigned tab of the Clerk of the Board organizational queue

Testing Plan
1. Go to Caseflow Login and Sign as BVADWISE
2. Navigate to Mail Intake
3. From the dropdown select VA FORM 10182 for processing and click on continue to
    search
4. Enter a veteran file number. I used 778820030. Click the search icon to continue
5. Fill out the form and select direct review for the review option type. Click the button at
    the bottom to continue
6. On the add issues page, click the add issues button. Select a benefit type as
    Compensation and Issue Category as "Contested claims - Attorney Fees. Click on Add
    this issue and establish appeal to complete the intake
7. The intake should be completed
8. Switch user and select Clerk of the board user and enter into Queue and click on Switch
    view and select Clerk of the board team view
<img width="830" alt="image" src="https://user-images.githubusercontent.com/110078646/223542157-1ccd8802-4588-4aa4-a655-119cc505722b.png">

9. The Send Final Notification Letter task is created in 2 ways. One is from Send Initial Notification Letter Task and the other
     one is from Post Send Initial Notification Letter Holding Task
10. From the Send Initial Notification Letter task you need to select and click on Mark as complete or Proceed to final letter.
     If you select Mark as complete, the Task will be created as Post Send Initial Notification Letter Holding Task.
     And the case is moved from "Unassigned tab" to "On hold tab"
     From Post Send Initial Notification Letter Holding Task you can select Proceed to final notification letter.
     Now the Send Final Notification Letter Task is created and the case moved from "On hold tab" to "Unassigned tab"
<img width="851" alt="image" src="https://user-images.githubusercontent.com/110078646/223543936-8dc8c867-33c9-4687-89cb-1beb0bfb2326.png">
<img width="830" alt="image" src="https://user-images.githubusercontent.com/110078646/223544180-a6b05c92-0e09-42bf-be3d-dd8b0f256d9d.png">

<img width="830" alt="image" src="https://user-images.githubusercontent.com/110078646/223544384-cc65c611-2038-4aa9-972a-7eef6bb53bcb.png">

<img width="818" alt="image" src="https://user-images.githubusercontent.com/110078646/223544526-402bf9d4-d5bd-4e46-8b1a-595390f248b7.png">

<img width="829" alt="image" src="https://user-images.githubusercontent.com/110078646/223545403-958564d8-b7c4-448d-ac67-b1268fbafdc2.png">

<img width="827" alt="image" src="https://user-images.githubusercontent.com/110078646/223547431-07028c4c-3af8-4e3a-b516-45a1d296dbf0.png">
<img width="838" alt="image" src="https://user-images.githubusercontent.com/110078646/223547546-ec87a40d-42b7-4007-bbe5-a9c9bdb9b67e.png">

<img width="808" alt="image" src="https://user-images.githubusercontent.com/110078646/223549293-1076c0d2-a420-4dd8-af11-058044eb15b1.png">

11. From the Send Initial Notification Letter, if you select and click on Proceed to final notification letter the task will be created as send Final Notification Letter. (Here I picked another case to go with the Contested Claim workflow)
<img width="827" alt="image" src="https://user-images.githubusercontent.com/110078646/223550118-82c491b9-e395-4ffd-b37c-ac6a12ca1bd0.png">

<img width="829" alt="image" src="https://user-images.githubusercontent.com/110078646/223550228-d5bb5f12-9f02-467b-a7b4-15a36bbb500d.png">
<img width="826" alt="image" src="https://user-images.githubusercontent.com/110078646/223550399-81bb358d-5e92-4759-9136-81e7ea4732f4.png">

<img width="800" alt="image" src="https://user-images.githubusercontent.com/110078646/223550571-20a68980-a967-40bb-bae0-fca6360ee2db.png">
